cloudbuild.yaml

```diff
@@ -22,148 +30,113 @@ steps:
-    export GITLAB_PASSWORD=$(kubectl get secrets gitlab-gitlab-initial-root-password -o jsonpath="{.data.password}" | base64 -d)
-    
-    # Create a personal access token with the value of the root password
-    export UNICORN_POD=$(kubectl get pods -l=app=unicorn -o jsonpath='{.items[0].metadata.name}')
-    export GITLAB_TOKEN=$${GITLAB_PASSWORD}
-    kubectl exec -it $$UNICORN_POD -c unicorn -- /bin/bash -c "
-    cd /srv/gitlab;
-    bin/rails r \"
-    token_digest = Gitlab::CryptoHelper.sha256 \\\"$${GITLAB_PASSWORD}\\\";
-    token=PersonalAccessToken.create!(name: \\\"Anthos Platform Installer\\\", scopes: [:api], user: User.where(id: 1).first, token_digest: token_digest);
-    token.save!
-    \";
-    " > /dev/null 2>&1 || true
-
-    # Add the GitLab instance's SSH host key
-    mkdir -p ~/.ssh
-    ssh-keyscan -t rsa gitlab.${_DOMAIN} >> ~/.ssh/known_hosts
-
-    export GITLAB_HOSTNAME=gitlab.${_DOMAIN}
-    git config --global user.email "anthos-platform-installer@example.com"
-    git config --global user.name "Anthos Platform Installer"
-
-    # Run create_repos.sh
-    export TERRAFORM_AUTO_APPROVE=true
-    ./create-repos.sh
-    cd ..
-
-    # Configure ACM CI
-
-    # Get an Oauth Token to authenticate with GitLab API
-    cat > auth.txt <<EOF
-    grant_type=password&username=root&password=$${GITLAB_PASSWORD}
-    EOF
-    export GITLAB_OAUTH_TOKEN=$(curl -k --data "@auth.txt" --request POST https://gitlab.${_DOMAIN}/oauth/token | jq -r .access_token)
-
-    # Get ACM repo project ID
-    export ACM_PROJECT_ID=$(curl -k -s "https://gitlab.${_DOMAIN}/api/v4/projects?search=anthos-config-management&access_token=$${GITLAB_OAUTH_TOKEN}" | jq -r .[0].id)
-
-    # Get runner registration token from GitLab
-    export ACM_RUNNER_TOKEN=$(curl -k -s "https://gitlab.${_DOMAIN}/api/v4/projects/$${ACM_PROJECT_ID}?access_token=$${GITLAB_OAUTH_TOKEN}" | jq -r .runners_token)
-
-    # Install ACM
-    cd 3_acm
-    ./install_acm.sh
-    cd ..
-
-    # Install demo apps Hipster Shop app (by microservice), and Petabank app
-    cd 4_demo
-    ./provision-applications.sh
-
-    # Add IAM policies to GSA service accounts
-    export PROJECT_ID=${PROJECT_ID}
-    ./add-gsa-iam-policies.sh
-
-    # Copy source code to repositories and trigger pipelines
-    ./build-applications.sh
-
-    # Modify default Network Policies
-    ./modify-network-policies.sh
-
-    # Delete apps created from initial pipeline run
-    ./delete-templates.sh
-    cd ..
-
-    # Output final notes
-    echo -e "\e[32mLog in to your GitLab instance at: https://gitlab.${_DOMAIN}"
-    echo -e "\e[32mUsername: root"
-    echo -e "\e[32mPassword: $${GITLAB_PASSWORD}"
-    echo
-    echo -e "\e[32mVisit the user guide in the docs (docs/index.md) to go through a user journey (add, deploy, and change applications)."
+    - -c
+    - |
+      export GITLAB_PASSWORD=$(gcloud secrets versions access latest --secret="gitlab-password")
```

Comment:
Rather than printing out the password in the build log, lets just print out how to pull it down (ie Retrieve Password)